src/internal/AnalyticsDispatcher.ts

```diff
@@ -0,0 +1,20 @@
+import {AnalyticsContext} from './AnalyticsContext';
+
+export type DispatchAnalytics<T> = (eventName: string, context: AnalyticsContext) => T;
+export type AnalyticsExtender<T> = (dispatcher: AnalyticsDispatcher<T>) => AnalyticsDispatcher<T>;
+
+export class AnalyticsDispatcher<T> {
+  constructor(
+    private readonly dispatchAnalytics: DispatchAnalytics<T>,
+    readonly context: AnalyticsContext = new AnalyticsContext(),
+  ) {}
+
+  dispatch = (eventName: string = '', analyticsContext?: AnalyticsContext): T => {
```

Comment:
Are `dispatch` and `extend` arrow functions instead of methods on purpose? Dispatchers are created a lot of times, this will cause each of them to create a new function